tests/test_multiaddr.py

```diff
@@ -218,3 +223,47 @@ def test_get_value():
     assert_value_for_proto(a, P_IP4, "0.0.0.0")
     assert_value_for_proto(a, P_UDP, "12345")
     assert_value_for_proto(a, P_UTP, "")
+
+
+def test_bad_initialization_no_params():
+    with pytest.raises(ValueError):
+        Multiaddr()
+
+
+def test_bad_initialization_too_many_params():
+    with pytest.raises(ValueError):
+        Multiaddr("/ip4/0.0.0.0", string_to_bytes("/ip4/0.0.0.0"))
+
+
+def test_get_value_too_many_fields_protocol(monkeypatch):
+    monkeypatch.setattr("multiaddr.multiaddr.Multiaddr.__str__",
+                        lambda ignore: '/udp/1234/5678')
```

Comment:
> The necessity of a monkeypatch here probably means I designed the Multiaddr class poorly. I'll keep this in the back of my mind for a refactor sometime later, but for now patching is a-ok.

Not sure I agree in general although maybe in this case.  This may very well be a code path that could never get hit in reality.